core/templates/domain/objects/RatioObjectFactorySpec.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for the ratio object type factory service
+ */
+
+import { ObjectsDomainConstants } from
+  'domain/objects/objects-domain.constants';
+import { Ratio, RatioObjectFactory } from
+  'domain/objects/RatioObjectFactory';
+
+describe('Ratio Object Factory', () => {
+  let errors = null;
+  let ratio: RatioObjectFactory = null;
+
+  beforeEach(() => {
+    errors = ObjectsDomainConstants.RATIO_PARSING_ERRORS;
+    ratio = new RatioObjectFactory();
+  });
+
+  it('should create a new object from list', () => {
+    const ratioObject = [1, 2, 3];
+    const createdRatio = ratio.fromList(ratioObject);
+
+    expect(createdRatio.numbers).toEqual(ratioObject);
+  });
+
+  it('should convert itself to a string in ratio format', () => {
+    expect(new Ratio([1, 2, 3]).toString()).toBe('1:2:3');
+    expect(new Ratio([2, 3, 5]).toString()).toBe('2:3:5');
+    expect(new Ratio([2, 4, 6]).toString()).toBe('2:4:6');
+    expect(new Ratio([10, 2, 15]).toString()).toBe('10:2:15');
+    expect(new Ratio([1, 2, 3, 4]).toString()).toBe('1:2:3:4');
+  });
+
+  it('should return the correct lenght of list', () => {
```

Comment:
```suggestion
  it('should return the correct length of list', () => {
```

Also, this doesn't quite read correctly. The full test would read "Ratio Object Factory should return the correct length of list" but what you're really testing is ``getNumberOfTerms``. See other test suites for an example. In cases like this, we usually cascade the describes to make the suite clearer, e.g.:

```typescript
describe('RatioObjectFactory', () => {
  describe('.getNumberOfTerms()', () => {
    it('should return the number of components', () => {
      ...
    })
  })
})
```

In this way, the test is contextualized.

Ditto for all other tests in this suite. Generic properties being evaluated can be top-level, but otherwise the API should be in the nearest 'describe'.

Reply:
done